thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AnomaliesResource.java

```diff
@@ -464,6 +467,69 @@ public void updateAnomalyMergedResultFeedback(@PathParam("mergedAnomalyId") long
     }
   }
 
+  /**
+   * Create a user-reported anomaly
+   *
+   * @param anomalyFunctionId anomaly function id (must exist)
+   * @param startTime start time utc (in millis)
+   * @param endTime end time utc (in millis)
+   * @param feedbackType anomaly feedback type
+   * @param comment anomaly feedback comment (optional)
+   * @param dimensionsJson dimension map json string (optional)
+   * @throws IllegalArgumentException if the anomaly function id cannot be found
+   * @throws IllegalArgumentException if the anomaly cannot be stored
+   */
+  @POST
+  @Path(value = "/reportAnomaly/{anomalyFunctionId}")
+  public void createUserAnomaly(@PathParam("anomalyFunctionId") long anomalyFunctionId,
+      @QueryParam("startTime") Long startTime,
+      @QueryParam("endTime") Long endTime,
+      @QueryParam("feedbackType") AnomalyFeedbackType feedbackType,
+      @QueryParam("comment") String comment,
+      @QueryParam("dimensionsJson") String dimensionsJson) {
+
+    AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(anomalyFunctionId);
+    if (anomalyFunction == null) {
+      throw new IllegalArgumentException(String.format("Could not resolve anomaly function id %d", anomalyFunctionId));
+    }
+
+    if (startTime == null) {
+      throw new IllegalArgumentException("Must provide startTime");
+    }
+
+    if (endTime == null) {
+      throw new IllegalArgumentException("Must provide endTime");
+    }
+
+    if (feedbackType == null) {
+      throw new IllegalArgumentException("Must provide feedbackType");
+    }
+
+    MergedAnomalyResultDTO anomaly = new MergedAnomalyResultDTO();
```

Comment:
Hi Alex, thanks for implementing this endpoint to take user-report anomaly. If there is already one (or more) anomaly overlap with user provided start and end time, shall we recover those as "user-report"? It will help on correcting tuning's training samples. Thanks.

Reply:
Yes, I remember that discussion. IMO we should decide recovery/merge/ignore when we prepare the data for auto-tuning. Otherwise, we risk losing the user input irrevocably by mingling it with algo outputs.

Reply:
Let me correct the purpose of having the match. Users' input time range can be arbitrary and can be interpreted as a guideline. In most of the case the start and end time of anomaly that user input can go beyond the region that the algorithm specify. User can have a rough impression to report (since in fact they don't usually know the accurate start time as well), thus we don't necessarily to take it as an accurate input. To make sure algorithm's anomaly removal + baseline prediction running properly, we need to 1) give user space to input anomalous region; 2) match user input region to existing anomalies (these anomalies are detected by the algorithm but just filtered out by alert-filter), following up with anomaly detection running, anomaly removal will be correctly done using the matched anomalies; 3) If no matches found,  run algorithm model selection (currently under design) to fully recover the user report. 

Overall, with user report anomaly, this is the only place to get matches, otherwise algorithm cannot pick up correct anomalies for removal, as well as auto-adjust the performance of algorithm.

Thanks!

Reply:
as per our discussion: https://github.com/linkedin/pinot/pull/2119